src/wasm/WasmModule.cpp

```diff
@@ -60,43 +94,59 @@ wasm::WasmEnvironment& WasmModule::getWasmEnvironment()
     return wasmEnvironment;
 }
 
-std::string WasmModule::snapshot()
+std::string WasmModule::snapshot(bool locallyRestorable)
 {
+    PROF_START(wasmSnapshot)
```

Comment:
I am wondering, should this `PROF_START` calls be merged in? I know they becom `no-op`s but I'd say these were useful for your debugging, but not to someone else who runs with trace?

Reply:
I would agree for some of the places I've put them in this PR should be removed as they're overkill, but I actually think having `PROF_START` and `PROF_END` around important, time-consuming stuff like snapshot and restore is very useful. Whenever someone switches this stuff on, they are going to be looking at a trace log, which will be huge to start with, so I wouldn't worry about a few extra lines. More importantly, they will be looking to explain what's taking time in their application, and this could be a source of latency that they weren't considering, so might pop out with something useful.